src/libraries/Microsoft.PowerFx.Core/Localization/IResourceStringManager.cs

```diff
@@ -7,10 +7,13 @@ namespace Microsoft.PowerFx.Core.Localization
     /// This interface is used by Canvas apps to pass in an interface to access strings for keys that are resolved later by PowerFx.
     /// Ideally it would be removed, but separating this is tricky, and this allows PowerFx to stand on its own. 
     /// </summary>
-    internal interface IExternalStringResources
+    [ThreadSafeImmutable]
+    internal interface IResourceStringManager
     {
         bool TryGetErrorResource(ErrorResourceKey resourceKey, out ErrorResource resourceValue, string locale = null);
 
         bool TryGet(string resourceKey, out string resourceValue, string locale = null);
+
+        public string Get(string resourceKey, string locale = null);
```

Comment:
> public string Get(string resourceKey, string locale = null); [](http://example.com/codeflow?start=7&length=61)

can we avoid  maybe make an extension method? 
interfaces should be minimal, so they should have redundant get and tryget.